website/content/en/docs/building-operators/golang/webhook.md

```diff
@@ -0,0 +1,288 @@
+│   │   ├── kustomization.yaml
+│   │   └── patches
+│   │       ├── basic.config.yaml
+│   │       └── olm.config.yaml
+│   └── webhook
+│       ├── kustomization.yaml
+│       ├── kustomizeconfig.yaml
+│       └── service.yaml
+├── controllers
+│   ├── memcached_controller.go
+│   └── suite_test.go
+├── go.mod
+├── go.sum
+├── hack
+│   └── boilerplate.go.txt
+└── main.go
+
+19 directories, 53 files
+```
+
+The scaffolded file `api/v1alpha1/memcached_webhook.go` has method signatures which need to be implemented for the validation webhook.
+
+The example memcached operator explained in the [tutorial](https://sdk.operatorframework.io/docs/building-operators/golang/tutorial/) is too simple to require defaulting or additional validations,
+but as an example, we can reinforce that the default value for `spec.size` should be `3`, by adding the following logic to 
+the `Default` function (note that this is already handled by the CRD defaulting and is technically completely superfluous):
+
+```sh
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *Memcached) Default() {
+	log.Info("default", "name", r.Name)
+
+	if r.Spec.Size == 0 {
+		r.Spec.Size = 3
+	}
+}
+```
+
+For validation, we can enforce that the size of the cluster follows a rule that is difficult or impossible 
+to describe with OpenAPI, for example, that the size of the cluster always remain an odd number.
+
+In order to perform this validation, we can simply add the `validateOdd` function shown below to  perform the check.
+
+```sh
+func validateOdd(n int32) error {
+	if n%2 == 0 {
+		return errors.New("Cluster size must be an odd number")
+	}
+	return nil
+}
+```
+
+At the end, let's call this method from `ValidateCreate` and `ValidateUpdate` function as shown below.
+
+```sh
+func (r *Memcached) ValidateCreate() error {
+	log.Info("validate create", "name", r.Name)
+	return validateOdd(r.Spec.Size)
+}
+
+func (r *Memcached) ValidateUpdate(old runtime.Object) error {
+	log.Info("validate update", "name", r.Name)
+	return validateOdd(r.Spec.Size)
+}
+```
+
+This function gets called whenever an object deletion happens.
+
+```sh
+func (r *Memcached) ValidateDelete() error {
+	log.Info("validate delete", "name", r.Name)
+
+	// TODO(user): fill in your validation logic upon object deletion.
+	return nil
+}
+```
+
+### Generate webhook manifests and enable webhook deployment 
+
+Once your webhooks are implemented, all that’s left is to create the WebhookConfiguration manifests required to register your webhooks with Kubernetes:
+
+```sh
+$ make manifests
+```
+
+You will need to enable cert-manager and webhook deployment in order to deploy these webhooks properly. 
+To do so, edit the `config/default/kustomize.yaml` and uncomment the sections marked by `[WEBHOOK]` and `[CERTMANAGER]` comments. 
+More detail on this step can be found in the [Kubebuilder documentation](https://book.kubebuilder.io/cronjob-tutorial/running-webhook.html).
```

Comment:
So what do we do after this? This feels incomplete. So i run `make manifests` I edit `kustomize.yaml` then what?

Reply:
Hereafter, we diverted user to the upstream doc as upstream doc has everything explained about updating **main.go** and next steps.

Reply:
```suggestion
```
This is not needed. Its covered in the second point above.